apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java

```diff
@@ -1503,4 +1819,157 @@ private void writeDebugLogMessages() {
+           * columnDefinition, so we do that. This leads to incorrect behavior with HSQL
+           * (for different reasons), but fortunately that doesn't happen to cause
+           * problems with our tests.
+           */
+          StringBuilder columnDefinition = new StringBuilder();
+          columnDefinition.append("numeric");
+          if (columnLength.isPresent() || columnScale.isPresent()) {
+            columnDefinition.append('(');
+            if (columnLength.isPresent()) {
+              columnDefinition.append(columnLength.get());
+            }
+            if (columnScale.isPresent()) {
+              columnDefinition.append(", ");
+              columnDefinition.append(columnScale.get());
+            }
+            columnDefinition.append(')');
+          }
+          columnAnnotation.addMember("columnDefinition", "$S", columnDefinition.toString());
+        }
+      }
+      annotations.add(columnAnnotation.build());
+    } else {
+      annotations.add(AnnotationSpec.builder(Transient.class).build());
+    }
+
+    return annotations;
+  }
+
+  private static TypeName selectJavaFieldType(
+      RifColumnType type,
+      boolean isColumnOptional,
+      Optional<Integer> columnLength,
+      Optional<Integer> columnScale) {
+    if (type == RifColumnType.CHAR
+        && columnLength.orElse(Integer.MAX_VALUE) == 1
+        && !isColumnOptional) return TypeName.CHAR;
+    else if (type == RifColumnType.CHAR
+        && columnLength.orElse(Integer.MAX_VALUE) == 1
+        && isColumnOptional) return ClassName.get(Character.class);
+    else if (type == RifColumnType.CHAR) return ClassName.get(String.class);
+    else if (type == RifColumnType.DATE && columnLength.orElse(0) == 8)
+      return ClassName.get(LocalDate.class);
+    else if (type == RifColumnType.TIMESTAMP && columnLength.orElse(0) == 20)
+      return ClassName.get(Instant.class);
+    else if (type == RifColumnType.NUM && columnScale.orElse(Integer.MAX_VALUE) > 0)
+      return ClassName.get(BigDecimal.class);
+    else if (type == RifColumnType.NUM
+        && columnScale.orElse(Integer.MAX_VALUE) == 0
+        && !isColumnOptional) return TypeName.INT;
+    else if (type == RifColumnType.NUM
+        && columnScale.orElse(Integer.MAX_VALUE) == 0
+        && isColumnOptional) return ClassName.get(Integer.class);
+    else throw new IllegalArgumentException("Unhandled field type: " + type.name());
+  }
+
+  private static TypeName selectJavaPropertyType(
+      RifColumnType type,
+      boolean isColumnOptional,
+      Optional<Integer> columnLength,
+      Optional<Integer> columnScale) {
+    if (!isColumnOptional)
+      return selectJavaFieldType(type, isColumnOptional, columnLength, columnScale);
+    else
+      return ParameterizedTypeName.get(
+          ClassName.get(Optional.class),
+          selectJavaFieldType(type, isColumnOptional, columnLength, columnScale));
+  }
```

Comment:
Can the pre-existing `selectJavaFieldType(...)` and `selectJavaPropertyType(...)` methods be refactored to call these under the hood? If so, we should definitely do that, to cut down on maintenance reqs.

Reply:
yes done..